core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -690,6 +704,7 @@ public void removeAll(final Set<? extends K> keys) throws BulkCacheWritingExcept
           resilienceStrategy.removeAllFailure(keys, e, new BulkCacheWritingException(failures, successes));
         }
       }
+      markAllCacheEventsFailed();
```

Comment:
This will never be applied if line 692 is executed. Consider moving it to a `finally` block.